simplex.py

```diff
@@ -170,6 +170,8 @@ def run(self, r_data=None):
                 ff.score = compare.compare_data(r_data, data)
                 opt.pretty_ff_results(ff)
         self.new_ffs = sorted(self.new_ffs + [ff_copy], key=lambda x: x.score)
+        # Allow 3 cycles w/o change for each parameter present
+        self.max_cycles_wo_change = 3*(len(self.new_ffs)-1)
```

Comment:
I followed the link. The discussion relates to variable names that you give names starting with underscore. This one doesn’t start with an underscore, therefore I didn’t consider it private. Non-private variables can be changed by convention.

Also, in our previous implementation, the number of allowed cycles without improvement must depend on the number of parameters included in the simplex, and you don’t know that at __init__ time.

From: Eric Hansen [mailto:notifications@github.com]
Sent: den 28 juni 2016 16:14
To: ericchansen/q2mm <q2mm@noreply.github.com>
Cc: Norrby, Per-Ola <Per-Ola.Norrby@astrazeneca.com>; Author <author@noreply.github.com>
Subject: Re: [ericchansen/q2mm] Mysimplex (#36)


In simplex.py<https://github.com/ericchansen/q2mm/pull/36#discussion_r68764184>:

> @@ -170,6 +170,8 @@ def run(self, r_data=None):

>                  ff.score = compare.compare_data(r_data, data)

>                  opt.pretty_ff_results(ff)

>          self.new_ffs = sorted(self.new_ffs + [ff_copy], key=lambda x: x.score)

> +        # Allow 3 cycles w/o change for each parameter present

> +        self.max_cycles_wo_change = 3*(len(self.new_ffs)-1)

This should not overwrite what is declared in __init__. This should rather be something like,

if not self.max_cycles_wo_change:

    self.max_cycles_wo_change = 3*(len(self.new_ffs)-1)

The other option is to change self.max_cycles_wo_change to self._max_cycles_wo_change, meaning that the user should not play with that setting. See this discussion<https://stackoverflow.com/questions/6930144/underscore-vs-double-underscore-with-variables-and-methods> for explanation.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ericchansen/q2mm/pull/36/files/0b9c5d0ad0a8d1105e4d26f192324c8bf323a928..398b4e1b8e1242703edfcc5b2a40b20047233942#r68764184>, or mute the thread<https://github.com/notifications/unsubscribe/ANVSNAcN9Wd3bApML33YdC_bSeC1kLmbks5qQSxDgaJpZM4JAC7r>.

________________________________

Confidentiality Notice: This message is private and may contain confidential and proprietary information. If you have received this message in error, please notify us and remove it from your system and note that you must not copy, distribute or take any action in reliance on it. Any unauthorized use or disclosure of the contents of this message is not permitted and may be unlawful.